tests/infra-validations.sh

```diff
@@ -241,5 +241,8 @@ git checkout $pr_id
 file_we_expect=("variables.tf" "main.tf" "backend.tfvars" "spk.tfvars")
 validate_directory "$TEST_WORKSPACE/$infra_generated_dir/$infra_hld_project-generated/$infra_region" "${file_we_expect[@]}" >> $TEST_WORKSPACE/log.txt
 echo "PR for generated repo validated."
-
+# Validate the contents of the definition.yaml
+spkVars_test=$'rg_name = "test-rg"\nrg_location = "west us2"\n'
+validate_file "$TEST_WORKSPACE/$infra_generated_dir/$infra_hld_project-generated/$infra_region/spk.tfvars" $spkVars_test >> $TEST_WORKSPACE/log.txt
```

Comment:
@NathanielRose I looked at `validate_file` and it is just a grep command with the quiet flag `grep -q "$2" "$1"` If there is more an $2 in the file it looks like we give success?

Reply:
@andrebriggs The reason why we use the grep command to check if a validation string is contained inside the file is due to line [127](https://github.com/CatalystCode/spk/blob/c0fcfec01f169d059cafe72f690855e14fd95484/tests/infra-validations.sh#L127). 

The definition yaml that gets created that we validate contains the PAT token and is a much larger file then the terraform file we validate at the end of the file. In the script, we check to see if a [validation string](https://github.com/CatalystCode/spk/blob/c0fcfec01f169d059cafe72f690855e14fd95484/tests/infra-validations.sh#L40) is within the definition.yaml. 

I can move forward to modify the function for `validate_file` to check the entire contents of the file and host it as a string in memory or a mock file. Or can I keep it as is with checking if a subset string exists in the file?

Reply:
IMO, this validation is weak and it is ok because the `definition.yaml` file is created by `spk infra scaffold` and there are already unit test for `generateClusterDefinition` function

Reply:
Agreed with @dennisseah but since we have a validation that checks the files of the PR, and our unit tests, I think this will suffice for checking the contents of the `spk.tfvars` file. We can revisit in the future.